### PR TITLE
[Core] Use inherited `__repr__` in raylet logging

### LINKED ISSUE #25817: [Core] Inherited `__repr__` is not respected in logging
### What happened + What you expected to happen

We show the name of the actor and the PID when redirecting logs from there. The displayed name of the actor can be overriden with the `__repr__` method. However, the logic in `_rayled.pyx` only checks if the class itself directly implements `__repr__` and does not respect `__repr__` that was inherited.

```python
                if (hasattr(actor_class, "__ray_actor_class__") and
                        "__repr__" in
                        actor_class.__ray_actor_class__.__dict__):  # will be false if __repr__ is inherited
                    actor_magic_token = "{}{}\n".format(
                        ray_constants.LOG_PREFIX_ACTOR_NAME, repr(actor))
                    # Flush on both stdout and stderr.
                    print(actor_magic_token, end="")
                    print(actor_magic_token, file=sys.stderr, end="")
```

I'd expect inheritance to work correctly here. I assume this could be achieved by changing the check to:

```python
                if (hasattr(actor_class, "__ray_actor_class__") and
                        actor_class.__ray_actor_class__.__repr__ != object.__repr__):
```

### Versions / Dependencies

090024c297e61170e3231aa2f7ca9d1a8ce7edfc

### Reproduction script

```python
import ray
import logging

logger = logging.getLogger(__name__)


class MyClass:
    def __repr__(self) -> str:
        return "ThisIsMyCustomActorName"

    def do(self):
        logger.warning("text")


class MySubclass(MyClass):
    pass


my_class_remote = ray.remote(MyClass)
my_subclass_remote = ray.remote(MySubclass)

my_class_actor = my_class_remote.remote()
ray.get(my_class_actor.do.remote())

my_subclass_actor = my_subclass_remote.remote()
ray.get(my_subclass_actor.do.remote())
```

Output:
```
(ThisIsMyCustomActorName pid=1239238) text
(MySubclass pid=1239275) text
```

Expected output:
```
(ThisIsMyCustomActorName pid=1239238) text
(ThisIsMyCustomActorName pid=1239275) text
```

### Issue Severity

Low: It annoys or frustrates me.

### PR DESCRIPTION
**Pull request by Yard1**
Signed-off-by: Antoni Baum <antoni.baum@protonmail.com>

## Why are these changes needed?

Changes the logic in `_raylet.pyx` to make sure that inherited `__repr__` is detected and used correctly in logging.

## Related issue number

Closes https://github.com/ray-project/ray/issues/25817

## Checks

- [x] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [x] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [x] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Comment by Yard1** [comment]
The train test is failing because we are not compiling wheels here (CI issue), working on fixing that in parallel

File: python/ray/_raylet.pyx [hunk 1]
```diff
@@ -920,8 +920,7 @@ cdef execute_task(
                 # log prefix with the full repr of the actor. The log monitor
                 # will pick up the updated token.
                 if (hasattr(actor_class, "__ray_actor_class__") and
-                        "__repr__" in
-                        actor_class.__ray_actor_class__.__dict__):
+                        actor_class.__ray_actor_class__.__repr__ != object.__repr__):
```
Comment (rkooo567) [inline python/ray/_raylet.pyx:923]: Dies __repr__ always guarantee to exist in all classes? Should we check the hasattr?

**Review by rkooo567 — APPROVED** [review]

File: python/ray/_raylet.pyx (hunk 1 above)
Comment (Yard1) [reply python/ray/_raylet.pyx:923]: Since all classes inherit from `object`, `__repr__` is always there.

**Comment by Yard1** [comment]
Test failure is the same as on master and unrelated.

**Pull request merged**